trl/trainer/training_configs.py

```diff
@@ -41,3 +41,33 @@ class RewardConfig(TrainingArguments):
     """If True, use gradient checkpointing to save memory at the expense of slower backward pass."""
     gradient_checkpointing_kwargs: Optional[dict] = None
     """Keyword arguments to pass to the gradient checkpointing function."""
+
+
+@dataclass
+class KTOConfig(TrainingArguments):
+    """
+    KTOConfig collects all training arguments related to the [`KTOTrainer`] class.
+
+    Using [`HfArgumentParser`] we can turn this class into
+    [argparse](https://docs.python.org/3/library/argparse#module-argparse) arguments that can be specified on the
+    command line.
+
+    Parameters:
+        max_length (`int`, *optional*, defaults to `None`):
+            The maximum length of the sequences in the batch. This argument is required if you want to use the default data collator.
+        max_prompt_length (`int`, *optional*, defaults to `None`):
+            The maximum length of the prompt. This argument is required if you want to use the default data collator.
+        max_target_length (`int`, *optional*, defaults to `None`):
+            The maximum length of the target. This argument is required if you want to use the default data collator and your model is an encoder-decoder.
+        beta (`float`, defaults to 0.1):
+            The beta factor in KTO loss. Higher beta means less divergence from the initial policy.
+    """
+
+    max_length: Optional[int] = None
+    """The maximum length of the sequences in the batch. This argument is required if you want to use the default data collator."""
+    max_prompt_length: Optional[int] = None
+    """The maximum length of the prompt. This argument is required if you want to use the default data collator."""
+    max_target_length: Optional[int] = None
```

Comment:
Just a suggestion to align with the expected dataset format (I realise this departs from `DPOTrainer` but maybe good to make it clearer with this new trainer):
```suggestion
    max_completion_length: Optional[int] = None
```